pulsar-broker/src/test/java/org/apache/pulsar/broker/s3offload/S3ManagedLedgerOffloaderTest.java

```diff
@@ -147,5 +221,28 @@ public void testNoBucketConfigured() throws Exception {
             // correct
         }
     }
+
+    @Test
+    public void testOffload() throws Exception {
+        int entryLength = 10;
+        int entryNumberEachBlock = 10;
+        ServiceConfiguration conf = new ServiceConfiguration();
+        conf.setManagedLedgerOffloadDriver(S3ManagedLedgerOffloader.DRIVER_NAME);
+        conf.setS3ManagedLedgerOffloadBucket(BUCKET);
+        conf.setS3ManagedLedgerOffloadRegion(REGION);
+        conf.setS3ManagedLedgerOffloadServiceEndpoint(endpoint);
+        conf.setS3ManagedLedgerOffloadMaxBlockSizeInBytes(
+            DataBlockHeaderImpl.getDataStartOffset() + (entryLength + 12) * entryNumberEachBlock);
+        LedgerOffloader offloader = new S3ManagedLedgerOffloader(conf, scheduler);
+
+        // offload 30 entries, which will be placed into 3 data blocks.
+        int entryCount = 30;
+        ReadHandle readHandle = buildReadHandle(entryCount);
+        UUID uuid = UUID.randomUUID();
+        offloader.offload(readHandle, uuid, new HashMap<>()).get();
+
+        S3Object obj = client.getObject(BUCKET, S3ManagedLedgerOffloader.offloadKey(readHandle, uuid));
+        verifyS3ObjectRead(obj, readHandle, 3, 30, conf.getS3ManagedLedgerOffloadMaxBlockSizeInBytes());
+    }
 }
```

Comment:
Add a test for the case where an individual ledger entry is larger than the configured block size.

Reply:
Also add tests for:

- The case where initMultipart fail
- the case where one of the uploadPart calls fails
- the case there the complete fails
- the case where putobject on the index fails

In all cases, we should clean up. Any exception on the multipart should trigger an abort multi part operation. Any other failure should trigger a deleteObjects on both the index and data object.

To trigger these failures you'll need to wait for https://github.com/apache/incubator-pulsar/pull/1755 to be in, and then use a Mockito spy to cause the failures.